[erv-teo] iP

erv-teo opened the pull request:
### Duke 
> "Your mind is for having ideas, not holding them." - David Allen [source](https://dansilvestre.com/productivity-quotes/)

Duke frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use 

All you need to do is,
1. download it from [here]()
2. double-click it 
3. add your tasks
4. let it manage your tasks for you 😍 

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
``` Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

erv-teo committed:
Level-0

erv-teo committed:
Level 1

erv-teo committed:
Level-2

erv-teo committed:
Level-3

erv-teo committed:
Level-4

erv-teo committed:
A-TextUiTesting

erv-teo committed:
Level-5

erv-teo committed:
Level-6

erv-teo committed:
A-Enums

TopKec commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+public class Deadline extends Task {
+    private String name;
+    private String by;
```
(extending this comment to the Event)
Might it be worth changing the date fields to things like `byDate`, `fromDate` etc? To more properly represent that these are indeed dates?

TopKec commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+public class Task {
+    private String name;
+    private boolean done;
```
Perhaps a change to `isDone` for the variable to more clearly communicate that it is a boolean variable?

AjayArvind2207 replied:
Seconding this!

TopKec commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,274 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+enum Commands {
```
Cool use of Enums to tidy up your code!

TopKec commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,274 @@
+            String[] arr = str.split(" ", 2);
+            String keyword = arr[0].toLowerCase();
+
+            switch (keyword) {
```
In line with conventions, might it make more correct to reduce the indentation on the `case`?

TopKec reviewed (commented):
Heya! Understand that this one's still a work in progress, but I've made some small comments on things here and there. Good luck with the project!

AjayArvind2207 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+    }
+
+    @Override
+    public String toString() {
+        if (this.done) {
+            return "[X] " + this.name;
+        } else {
+            return "[ ] " + this.name;
+        }
+    }
```
Alternative suggestion: maintain another method to generate either the [X] or [ ], since the this.name is redundant with both the conditions

AjayArvind2207 reviewed (commented):
Well done with your code! I loved the use of enums and your code is very succinct. In line with the other reviewer I believe there are some basic coding style violations you can fix, such as the naming of boolean variables. I look forward to seeing more of your progress!

erv-teo committed:
Level 7

erv-teo committed:
Merge branch 'branch-Level-7'

erv-teo committed:
Save marked data

erv-teo committed:
Merge branch 'branch-Level-7'

erv-teo committed:
Added new functions in Task

erv-teo committed:
Merge branch 'branch-Level-7'

erv-teo committed:
Level 8

erv-teo committed:
A-MoreOOP

erv-teo committed:
Added hello() to Ui, removed execute() in Command

erv-teo committed:
Placed classes in one package

erv-teo committed:
Data

erv-teo committed:
Merge branch 'master' into add-gradle-support

erv-teo committed:
Gradle

erv-teo committed:
JUnit tests for Parser and Ui

erv-teo committed:
Javadocs

erv-teo committed:
Coding Standard

erv-teo committed:
Level-9

erv-teo committed:
Merge branch 'branch-Level-9'

erv-teo committed:
Added header comments in Command

erv-teo committed:
Change variable names to match style guide

MarcusGitty commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    public void run() throws DukeException{
+        this.storage.createFile();
+        this.storage.load(tasks);
+
+        ui.hello();
```
I like how you named the methods, it's very clear what is happening

MarcusGitty commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,167 @@
+                break;
+            case "e":
+            case "event":
+                String[] frarr = arr[1].split("/from ", 2);
+                if (frarr.length <= 1) {
+                    throw new DukeException("Please use the format: event /from <date/time> /to <date/time>");
+                }
+                String[] toarr = frarr[1].split("/to ", 2);
+                if (toarr.length <= 1) {
+                    throw new DukeException("Please use the format: event /from <date/time> /to <date/time>");
+                }
+                String name1 = frarr[0];
+                String fromStr = toarr[0].trim();
+                String toStr = toarr[1].trim();
+                LocalDateTime from;
+                LocalDateTime to;
+                try {
+                    from = LocalDateTime.parse(fromStr, dtf);
+                    to = LocalDateTime.parse(toStr, dtf);
+                    task = new Event(name1, from, to);
+                } catch (DateTimeParseException e) {
+                    try {
+                        from = LocalDateTime.parse(fromStr + currentYear, dtf);
+                        to = LocalDateTime.parse(toStr + currentYear, dtf);
+                        task = new Event(name1, from, to);
+                    } catch (DateTimeParseException f) {
+                        try {
+                            from = LocalDateTime.parse(fromStr + currentYear + DEFAULT_START, dtf);
+                            to = LocalDateTime.parse(toStr + currentYear + DEFAULT_END, dtf);
+                            task = new Event(name1, from, to);
+                        } catch (DateTimeParseException g) {
+                            throw new DukeException("Invalid time format. Please use ddMM, ddMMyy HHmm, or ddMM HHmm");
+                        }
+                    }
+
+                }
+
+                break;
+            default:
+                throw new DukeException("Invalid format");
```
The case methods seems long to be, do you think it would be better to refactor to something for e.g

case "event":
     event.handler();

MarcusGitty commented on src/main/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,28 @@
+    @Test
+    public void multipleInputDeadlineSuccess() throws Exception {
+
+        // ddMM
+        assertEquals(new Deadline("iP ", ldt).toString(), parseFromInput("d iP /by 2202").toString());
+        // ddMMyy HHmm
+        assertEquals(new Deadline("iP ", ldt).toString(), parseFromInput("d iP /by 220224 2359").toString());
+        // ddMM HHmm
+        assertEquals(new Deadline("iP ", ldt).toString(), parseFromInput("d iP /by 2202 2359").toString());
```
I like how you tested for different cases.

MarcusGitty reviewed (commented):
Overall I understand that this is still a work in progress, all the best for your IP buddy, I think we are on the same pace, JIAYOUS !!

erv-teo committed:
Implement GUI for Duke

erv-teo committed:
Implement duke functionality with GUI

erv-teo committed:
Change DialogBox and MainWindow to display longer texts

erv-teo committed:
Shorten methods in Parser.java, change MainWindow.java to display welcome message upon start

erv-teo committed:
Use assert feature to ensure file is created when the application is run

erv-teo committed:
Merge pull request #2 from erv-teo/branch-A-Assertions

Use assert feature to ensure file is created when the application is run

erv-teo committed:
Add Javadocs to Duke#getResponse()

erv-teo committed:
Change code to adhere to code quality standards

erv-teo committed:
Merge branch 'branch-A-CodeQuality'

erv-teo committed:
Add tagging feature

Tagging feature allows users to tag tasks with a
specific tag.
A tag is implemented as a new class and added as a
field to each task.

erv-teo committed:
Update README.md

erv-teo committed:
Rename the project to Patrick

Add title to stage so that it is not empty.

Add Ui.png in docs/ to showcase the UI of the
project.

erv-teo committed:
Update README.md

erv-teo committed:
Update README.md
